selfdrive/locationd/helpers.py

```diff
@@ -70,8 +70,12 @@ def get_msg(self, valid: bool, with_points: bool) -> log.Event:
 
 def cache_points_onexit(param_name, estimator, sig, frame):
   signal.signal(sig, signal.SIG_DFL)
+  cache_points(param_name, estimator, True)
+  sys.exit(0)
+
+
+def cache_points(param_name, estimator, valid):
   cloudlog.warning(f"Caching {param_name} param")
   params = Params()
-  msg = estimator.get_msg(valid=True, with_points=True)
+  msg = estimator.get_msg(valid=valid, with_points=True)
   params.put(param_name, msg.to_bytes())
```

Comment:
needs to use put_nonblocking if we're writing while onroad, otherwise you're vulnerable to slow disk io

Reply:
Updated.